src/phasorpy/_phasorpy.pyx

```diff
@@ -1855,3 +1857,88 @@ cdef float_t _anscombe_inverse_approx(
         + 0.7654655446197431 / (x * x * x)  # 5/8 * sqrt(3/2)
         - 0.125  # 1/8
     )
+
+
+###############################################################################
+# Filtering functions
+
+cdef float_t _quickselect(float_t *arr, int left, int right, int k) nogil:
```

Comment:
- All `int` types for indices and sizes should be replaced by `ssize_t` type in order to be able to process > 2 GB images.
- `nogil` -> `noexcept nogil`
- Use `const` qualifiers if possible.

`cdef float_t _quickselect(const float_t *arr, ssize_t left, ssize_t right, ssize_t k) noexcept nogil:`

Reply:
I tried using the `const` qualifier with `*arr` but it doesn;t seem to be compatible with `nogil`

Reply:
I'll look into `const float_t *arr`. Maybe use memoryviews instead.

`k` seems const too?

Reply:
You are right, `k` is also `const`. 

I realized that `*arr` is not `const` since the algorithm changes the order of the elements to sort them. For example:

`arr[pivot_new_index] = temp`